app/controllers/api/physical_storages_controller.rb

```diff
@@ -29,5 +29,21 @@ def delete_resource_action(type, id = nil, _data = nil)
         {:task_id => physical_storage.delete_physical_storage_queue(User.current_user)}
       end
     end
+
+    def validate_resource(type, id = nil, data = {})
+      api_action(type, id) do |physical_storage|
+        raise BadRequestError, 'Validate Physical Storage API must get provider id' unless data.key?('ems_id')
+
+        ems = resource_search(data['ems_id'], :providers)
+
+        raise BadRequestError, "Couldn't find a provider by the provider id - '#{data['ems_id']}'" unless ems
+
+        klass = ems.class_by_ems(physical_storage.name.split(':').last)
+        ensure_supports(type, klass, :validate)
+
+        task_id = physical_storage.validate_storage_queue(User.current_userid, ems, data)
+        action_result(true, "Validating #{physical_storage.name}", :task_id => task_id)
+      end
```

Comment:
This looks too similar to `create_ems_resource` (thank you for that)

I am introducing `api_ems_resource` via #1167 for your use.

```suggestion
      api_ems_resource(type, data, "Validating", data, :supports => :validate) do |ems, klass|
        {:task_id => physical_storage.validate_storage_queue(User.current_userid, ems, data)}
      end
```

Reply:
thanks, Keenan, great solution.
I'll test it and push.

Reply:
@kbrock I've changed the code to use api_ems_resource.
Please, before you merge your code to master, read my comment:
https://github.com/ManageIQ/manageiq-api/pull/1167/files#r901095007
Thanks